features/git-rename-branch/feature_branch/with_remote_origin/with_tracking_branch/with_unpulled_changes.feature

```diff
@@ -0,0 +1,33 @@
+Feature: git rename-branch: errors if renaming a feature branch that has unpulled changes
+
+  As a developer renaming a feature branch that has unpulled changes
```

Comment:
Reading this again, it becomes clearer to me that we definitely must implement the branch syncing into this command (in a separate PR). Otherwise, without running `git fetch`, how do we even know whether this branch has unpulled changes?